File: parent335/pinyougou-interface/src/main/java/cn/itcast/core/service/brand/BrandService.java
package cn.itcast.core.service.brand;

import cn.itcast.core.entity.PageResult;
import cn.itcast.core.pojo.good.Brand;

import java.util.List;
import java.util.Map;

public interface BrandService {
    /**
     * 查询所有的品牌
     * @return
     */
    public List<Brand> findAll();

    /**
     * 品牌管理结果分页
     * @param pageNo
     * @param pageSize
     * @return
     */
    public PageResult findPage(Integer pageNo,Integer pageSize);

    /**
     * 条件查询
     * @param pageNo
     * @param pageSize
     * @param brand
     * @return
     */
    public PageResult search(Integer pageNo,Integer pageSize,Brand brand);

    /**
     * 保存品牌
     */
    public void add(Brand brand);

    /**
     * 回显品牌
     */
    public Brand findOne(Long id);

    /**
     * 更新品牌
     */
    public void update(Brand brand);

    /**
     * 删除
     * @param ids
     */
    public void delete(Long[] ids);

    /**
     * 新增模板时需要加载的品牌结果集
     * @return
     */
    List<Map> selectOptionList();
/*
* 审核品牌
* */
    void updateStatus(Long[] ids, String status);
}

File: parent335/pinyougou-web-manager/src/main/webapp/js/service/brandService.js
// 定义服务层:
app.service("brandService",function($http){
    this.findAll = function(){
        return $http.get("../brand/findAll.do");
    }

    this.findByPage = function(page,rows){
        return $http.get("../brand/findByPage.do?page="+page+"&rows="+rows);
    }

    this.save = function(entity){
        return $http.post("../brand/add.do",entity);
    }

    this.update=function(entity){
        return $http.post("../brand/update.do",entity);
    }

    this.findById=function(id){
        return $http.get("../brand/findOne.do?id="+id);
    }

    this.dele = function(ids){
        return $http.get("../brand/delete.do?ids="+ids);
    }

    this.search = function(page,rows,searchEntity){
        return $http.post("../brand/search.do?pageNo="+page+"&pageSize="+rows,searchEntity);
    }

    this.selectOptionList = function(){
        return $http.get("../brand/selectOptionList.do");
    }

    //导入Excel
    this.uploadExcel = function() {
        // 向后台传递数据:
        var formData = new FormData();
        // 向formData中添加数据:
        var file = document.querySelector('input[type=file]').files[0];
        formData.append('file', file);

        return $http({
            method: 'post',
            url: '../brand/uploadExcel.do',
            data: formData,
            headers: {'Content-Type': undefined},// Content-Type : text/html  text/plain
            transformRequest: angular.identity
        });
    }



    this.updateStatus = function(ids,status){
        return $http.get('../brand/updateStatus.do?ids='+ids+"&status="+status);
    }
});